lib/document.js

```diff
@@ -541,94 +546,29 @@ class Document extends Readable {
     // close readable stream
     this.push(null)
   }
-}
-
-class AliasGenerator {
-  constructor() {
-    this.nextId = {}
-    this.blocked = new Set()
-  }
-
-  next(prefix) {
-    if (!(prefix in this.nextId)) {
-      this.nextId[prefix] = 1
-    }
-
-    let next
-    do {
-      next = prefix + this.nextId[prefix]++
-    } while (this.blocked.has(next))
-
-    return next
-  }
-
-  block(alias) {
-    alias = String(alias)
-    if (alias[0] === '/') {
-      alias = alias.slice(1)
-    }
-
-    this.blocked.add(alias)
-  }
-
-  isBlocked(alias) {
-    alias = String(alias)
-    if (alias[0] === '/') {
-      alias = alias.slice(1)
-    }
-
-    return this.blocked.has(alias)
-  }
-
-  reset(prefix) {
-    this.nextId[prefix] = 1
-  }
-}
-
-module.exports = class extends Fragment {
-  constructor(opts) {
-    const doc = new Document(opts)
-    super(doc, doc)
-  }
-
-  get info() {
-    return this._doc.info
-  }
-
-  pipe(dest, opts) {
-    return this._doc.pipe(dest, opts)
-  }
-
-  on() {
-    return this._doc.on.apply(this._doc, arguments)
-  }
-
-  async end() {
-    await Fragment.prototype.end.call(this)
-    await this._doc.end()
-  }
 
   asBuffer(callback) {
     let p = new Promise((resolve, reject) => {
       const chunks = []
-      this.on('data', chunk => chunks.push(chunk))
-      this.on('end', () => resolve(Buffer.concat(chunks)))
-      this.on('error', reject)
-      this.end()
+      this._rootFragment.on('data', chunk => chunks.push(chunk))
+      this._rootFragment.on('end', () => resolve(Buffer.concat(chunks)))
+      this._rootFragment.on('error', reject)
+      this._rootFragment.end()
```

Comment:
I think they should still be `this.on` instead of `this._rootFragment.on`, but I am not 100% sure - I guess I am missing a test case for `asBuffer` here

Reply:
You are right on this, I have added a test for `asBuffer` as well.